File: Exercises/ExercisesC/11Jan_Chapter07/4-BarrierCrossing/Source/force.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
 
// calculate the forces and potential energy
void Force(double x, double *U, double *F)
{
  *U=0.0;
  *F=0.0;

  if(x>=-1.0&&x<=1.0)
  {
    *U=0.5*(1.0+cos(M_PI*x));
    *F=0.5*M_PI*sin(M_PI*x);
  }
} 

File: Exercises/ExercisesFortran/11Jan_Chapter07/4-BarrierCrossing/Source/force.f
      Subroutine Force(X,U,F)
      Implicit None
 
C     Calculate The Forces And Potential Energy
 
      Include 'system.inc'
 
      Double Precision X,U,F

      U = 0.0d0
      F = 0.0d0

      If(X.Ge.-1.0d0.And.X.Le.1.0d0) Then
         U =  0.5d0*(1.0d0 + Dcos(Onepi*X))
         F =  0.5d0*Onepi*Dsin(Onepi*X)
      Endif

      Return
      End
